app/src/main/java/com/fastlink/zekrapp/ui/utils/ZekirCategoryViewModelInterface.kt

```diff
@@ -0,0 +1,5 @@
+package com.fastlink.zekrapp.ui.utils
+
+interface ZekirCategoryViewModelInterface {
```

Comment:
Just because you have a function used in two places doesn't mean that you should create an interface for it. Remove this.

Reply:
I didn't create this interface for the updating function. We could implement this function in both ViewModels without an interface. However, I created this interface because, in the `CategoryItemCard` file we have `CategoryItem` composable function, we have a `viewModel` parameter that must accept both `HomeScreenViewModel` and `FavoriteScreenViewModel` since we use this composable in both the `favorite screen` and the `home screen`. There are two ways to handle this: either create a sealed class and extend it with both ViewModels, or create an interface, implement it in both ViewModels, and set the `viewModel` parameter type to this interface to accept both `HomeScreenViewModel` and `FavoriteScreenViewModel`

Reply:
That's not how you make your composables reusable in different screens. Rather you should have a callback function in the parameter that handles what will happen when the icon is clicked.